ksql-engine/src/main/java/io/confluent/ksql/connect/ConnectPollingService.java

```diff
@@ -0,0 +1,158 @@
+  private static final int INTERVAL_S = 30;
+
+  private final KsqlExecutionContext executionContext;
+  private final Consumer<CreateSource> sourceCallback;
+
+  private Set<Connector> connectors;
+
+  ConnectPollingService(
+      final KsqlExecutionContext executionContext,
+      final Consumer<CreateSource> sourceCallback
+  ) {
+    this.executionContext = Objects.requireNonNull(executionContext, "executionContext");
+    this.sourceCallback = Objects.requireNonNull(sourceCallback, "sourceCallback");
+    this.connectors = ConcurrentHashMap.newKeySet();
+  }
+
+  /**
+   * Add this connector to the set of connectors that are polled by this
+   * {@code ConnectPollingService}. Next time an iteration is scheduled in
+   * {@value #INTERVAL_S} seconds, the connector will be included in the topic
+   * scan.
+   *
+   * @param connector a connector to register
+   */
+  void addConnector(final Connector connector) {
+    connectors.add(connector);
+  }
+
+  @Override
+  protected void runOneIteration() {
+    if (connectors.isEmpty()) {
+      // avoid making external calls if unnecessary
+      return;
+    }
+
+    try {
+      final Set<String> topics = executionContext.getServiceContext()
+          .getAdminClient()
+          .listTopics()
+          .names()
+          .get(10, TimeUnit.SECONDS);
+
+      final Set<String> subjects = ImmutableSet.copyOf(
+          executionContext.getServiceContext()
+              .getSchemaRegistryClient()
+              .getAllSubjects()
+      );
+
+      for (final String topic : topics) {
+        final Optional<Connector> maybeConnector = connectors
+            .stream()
+            .filter(candidate -> candidate.matches(topic))
+            .findFirst();
+        maybeConnector.ifPresent(connector -> handleTopic(topic, subjects, connector));
+      }
+    } catch (final Exception e) {
+      LOG.error("Could not resolve connect sources. Trying again in {} seconds.", INTERVAL_S, e);
+    }
+  }
+
+  private void handleTopic(
+      final String topic,
+      final Set<String> subjects,
+      final Connector connector
+  ) {
+    final String valueSubject = topic + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
+    if (subjects.contains(valueSubject)) {
+      final String name = connector.getName();
+      final String source = connector.mapToSource(topic).toUpperCase();
+
+      // if the meta store already contains the source, don't send the extra command
+      // onto the command topic
+      if (executionContext.getMetaStore().getSource(source) == null) {
+        LOG.info("Auto-Registering topic {} from connector {} as source {}",
+            topic, connector.getName(), source);
+        final Builder<String, Literal> builder = ImmutableMap.<String, Literal>builder()
+            .put(CommonCreateConfigs.KAFKA_TOPIC_NAME_PROPERTY, new StringLiteral(topic))
+            .put(CommonCreateConfigs.VALUE_FORMAT_PROPERTY, new StringLiteral("AVRO"))
```

Comment:
Are we always import data in `AVRO` format? What if we want other formats?

Reply:
For now, yes, since Schema Registry only supports AVRO. When it supports other types hopefully there would be some API that allows me to get what format it is and inject it here.